frontend/src/features/projects/spl/steps/SplStep.tsx

```diff
@@ -153,14 +153,14 @@ const SplStep = ({ formikRef }: IStepProps) => {
       <Formik
         initialValues={initialValues}
         validateOnMount={true}
-        enableReinitialize={true}
```

Comment:
So, we've had a few errors caused by this over the course of this project and so I'm starting to think we need to avoid using this in the future.

Basically what this does is reinitialize the entire formik form with the passed initialValues whenever anything above the formik form in the render tree changes. This causes all unsaved changes on the form to be wiped, which is what is happening in this case (all unsaved changes are being lost right before the form is submitted, or whenever the tab changes).

Generally there should be nothing above the formik form changing in the render tree so this is usually safe, but this assumption has proven to be brittle in this and other instances so I think we should try and avoid using it from now on to avoid situations like this and in the future.

The alternative is to manually update the form values with formik's setValues method. This ensures that you only overwrite the form with an explicit call, which is nice. It does mean that we need to be careful to remember to call setValues, or the form will end up out of sync with the DB causing concurrency exceptions. Overall I think those end up being easier to debug, so I think I prefer this approach.

Reply:
I would have taken days to debug this...  Thanks!